app/helpers/route_overrides_helper.rb

```diff
@@ -0,0 +1,60 @@
+module RouteOverridesHelper
+  class << self
+    private
+
+    # @param format [Symbol] Either :singular or :plural
+    def mapping(format, options)
+      if format == :singular
+        from = options[:from].to_s.singularize
+        to = options[:to].to_s.singularize
+      else
+        from = options[:from].to_s.pluralize
+        to = options[:to].to_s.pluralize
+      end
+      mapping = {}
+      mapping[from] = to
+      mapping
+    end
+
+    def mapping_for(options)
+      should_map_singular = options[:singular].nil? || options[:singular]
+      should_map_plural = options[:plural].nil? || options[:plural]
+
+      singular_mapping = should_map_singular ? mapping(:singular, options) : {}
+      plural_mapping = should_map_plural ? mapping(:plural, options) : {}
+      singular_mapping.merge(plural_mapping)
+    end
+
+    def override_route_helpers_with_mapping(mapping)
```

Comment:
what is the difference between `def foo(options)` and `def foo(**options)`? In both cases, we can call the function with `foo(a: 1, b: 2)`